TCAT/Controllers/RouteDetail+DrawerViewController.swift

```diff
@@ -230,62 +231,55 @@ class RouteDetailDrawerViewController: UIViewController, UITableViewDataSource,
     /// Fetch delay information and update table view cells.
     @objc func getDelays() {
         
-        let firstDepartDirection = directions.first(where: { $0.type == .depart })
+        // First depart direction
+        guard let delayDirection = directions.first(where: { $0.type == .depart }) else {
+            return
+        }
         
-        /// Variable to make sure a nil direction being set to the delay isn't reset.
-        var shouldReset: Bool = true
+        directions.forEach { $0.delay = nil }
         
-        // Update delay variable of directions not directly related to transit.
-        func updateRemainingDirections(with optionalDelay: Int?) {
-            directions.filter { $0.type == .walk || $0.type == .arrive }.forEach { (direction) in
-                
-                // If no delay, nil the delay
-                guard let delay = optionalDelay else {
-                    direction.delay = nil
-                    return
-                }
-                
-                // Delay exists
+        let tripID = delayDirection.tripIdentifiers?.first
+        var stopID = delayDirection.stops.first?.id
+        
+        // Retrieve deleted stop ID (Route Parse) where first and last directions are removed.
+        if tripID != nil && stopID == nil {
+            stopID = route.rawDirections.first { (rawDir) -> Bool in
+                return rawDir.tripIdentifiers?.first == tripID
+            }?.stops.first?.id
+        }
+        
+        if let tripId = tripID, let stopId = stopID {
+            
+            Network.getDelay(tripId: tripId, stopId: stopId).perform(withSuccess: { (json) in
                 
-                // Direction has existing delay
-                if direction.delay != nil {
-                    // Reset delay to accumulate new results
-                    if shouldReset { direction.delay = 0 }
-                    direction.delay! += delay
-                }
+                if json["success"].boolValue {
                     
-                    // Direction doesn't have a delay
+                    // print("Got delay of \(json["data"]["delay"].int ?? -1), reloading data")
+                    delayDirection.delay = json["data"]["delay"].int
+                    
+                    // Update delay variable of other ensuing directions
+                    
+                    self.directions.filter {
+                        let isAfter = self.directions.index(of: delayDirection)! < self.directions.index(of: $0)!
+                        return isAfter && $0.type != .depart
+                        }.forEach { (direction) in
```

Comment:
put `.forEach` on new line/separate it from the `.filter`. Makes it easier to read